collectors/auth0/auth0_collector.js

```diff
@@ -0,0 +1,105 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2019, Alert Logic, Inc
+ * @doc
+ *
+ * Auth0 collector class.
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+'use strict';
+
+const moment = require('moment');
+const request = require('request');
+const parse = require('@alertlogic/al-collector-js').Parse;
+const ManagementClient = require('auth0').ManagementClient;
+const PawsCollector = require('@alertlogic/paws-collector').PawsCollector;
+
+
+const typeIdPaths = [
+    { path: ['type'] }
+];
+
+const tsPaths = [
+    { path: ['date'] }
+];
+
+
+class Auth0Collector extends PawsCollector {
+    constructor(context, creds) {
+        super(context, creds, 'auth0');
+    }
+    
+    pawsInitCollectionState(event, callback) {
+        const initialState = {
+            since: process.env.paws_collection_start_ts ? process.env.paws_collection_start_ts : moment().subtract(5, 'minutes').toISOString(),
+            poll_interval_sec: 1
+        };
+        return callback(null, initialState, 1);
+    }
+    
+    pawsGetLogs(state, callback) {
+        let collector = this;
+        const auth0Client = new ManagementClient({
+            domain: process.env.paws_domain,
+            clientId: process.env.paws_client_id,
+            clientSecret: process.env.paws_api_secret,
+            scope: 'read:logs'
+        });
+        let params = state.last_log_id ? {from: state.last_log_id} : {q: "date=[" + state.since + " TO *]"};
```

Comment:
Looks like `state.since` is not updated in `_getNextCollectionState()`. Is that ok?

Reply:
yes, it is only used for the initial fetch, after that we always fetch based on a `last_log_id`.